mender-auth/cli/actions.cpp

```diff
@@ -80,6 +82,39 @@ error::Error DoBootstrap(shared_ptr<MenderKeyStore> keystore, const bool force)
 			return err;
 		}
 	}
+	return err;
+}
+
+error::Error DoAuthenticate(context::MenderContext &main_context) {
+	events::EventLoop loop;
+	auto &config = main_context.GetConfig();
+	if (config.server_url == "") {
+		log::Info("No server set in the configuration, skipping authentication");
+		return error::NoError;
+	}
+	http::ClientConfig client_config {
+		config.server_certificate, config.https_client.certificate, config.https_client.key};
+	http::Client client {client_config, loop};
+	auto err = auth_client::FetchJWTToken(
+		client,
+		config.server_url,
+		config.paths.GetKeyFile(),
+		config.paths.GetInventoryScriptsDir(),
+		[&loop, &config](auth_client::APIResponse resp) {
+			log::Info("Got Auth response");
+			if (resp) {
+				log::Info(
+					"Successfully authorized with the server: " + config.server_url + resp.value());
+			} else {
+				log::Error(resp.error().String());
+			}
+			loop.Stop();
+		},
+		config.tenant_token);
+	if (err != error::NoError) {
+		return err;
+	}
+	loop.Run();
```

Comment:
Shouldn't we have a timer in parallel for a `loop.Stop` after a sensible timeout? Could it happen that this hangs for very long? :thinking:

Reply:
It is a very valid point imo! 

Tbh, I'm a little scared of this for our main loop.

In this instance, we could probably have a timeout on the loop, like we do for our test-loop, just to be sure.

In any case, this is supposed to be ran from the CLI, so if it hangs, you'd expect the user to stop it.

Alas, I agree. What is a sensible timeout you think? `5 sec` ? :thinking:

Reply:
As you said the user is most likely going to cancel, so for this kind of timeout I would say 30 seconds.

Question: do we print a info message like "authenticating to https:/..."? This would help for the user to cancel if it takes long.